feat: celestia-app docker document (#1364)
celestiaorg/docs: nodes/docker-images.md

---

> Sidebar_Label: Docker images

Reviewer: ```suggestion
sidebar_Label: Docker images
```

> description: Comprehensive guide for setting up Celestia-App and Celestia-Node using Docker.

Reviewer: ```suggestion
description: Comprehensive guide for setting up celestia-app and celestia-node using Docker.
```

> Description: Comprehensive guide for setting up Celestia-App and Celestia-Node using Docker.

Reviewer: ```suggestion
description: Comprehensive guide for setting up Celestia-App and Celestia-Node using Docker.
```

---

> # Celestia-App and Celestia-Node Setting Up with Docker 🐳

Reviewer: ```suggestion
# Celestia-app and celestia-node setup with Docker 🐳
```

import mainnetVersions from '/.vitepress/constants/mainnet_versions.js'
</script>

## Introduction

This page provides instructions on how to set up both the Celestia-App and
the Celestia-Node using Docker.
see [celestia-node page](./celestia-node.md) for additional instructions

> to run celestia-node using Docker.

Reviewer: sure but this is incorrect anyways

Docker is a container technology that offers a seamless method
for running applications in isolated environments on any
operating system.

This means that you can run Celestia-app & celestia-node
on your machine without having to worry about installing
and configuring all the dependencies required.

If you would like to learn more about
key management in Docker, visit the
[Docker and `cel-key` section](../developers/celestia-node-key.md#docker-and-cel-key).

## Docker Installation

To install Docker on your machine, see the links below for instructions
on your specific OS.

- [Docker Desktop for Mac or Windows](https://docs.docker.com/get-docker) or

- [Docker Engine for Linux](https://docs.docker.com/engine/install/)
  

> - Ensure you have a basic understanding of Docker to proceed.

Reviewer: sure

import mainnetVersions from '/.vitepress/constants/mainnet_versions.js'
</script>

## Introduction

> This page provides instructions on how to set up both the Celestia-App and

Reviewer: - [ ] use celestia-app and celestia-node throughout this document

> This page provides instructions on how to set up both the Celestia-App and

Reviewer: ```suggestion
This page provides instructions on how to set up both the celestia-app and
```

> the Celestia-Node using Docker.

Reviewer: ```suggestion
the celestia-node using Docker.
```

> see [celestia-node page](./celestia-node.md) for additional instructions

Reviewer: **no it is on this page, is this AI-generated??**

to run celestia-node using Docker.

Docker is a container technology that offers a seamless method
for running applications in isolated environments on any
operating system.

> This means that you can run Celestia-app & celestia-node

Reviewer: ```suggestion
This means that you can run celestia-app & celestia-node
```

on your machine without having to worry about installing
and configuring all the dependencies required.

If you would like to learn more about
key management in Docker, visit the
[Docker and `cel-key` section](../developers/celestia-node-key.md#docker-and-cel-key).

## Docker Installation

To install Docker on your machine, see the links below for instructions
on your specific OS.

- [Docker Desktop for Mac or Windows](https://docs.docker.com/get-docker) or

- [Docker Engine for Linux](https://docs.docker.com/engine/install/)
  
- Ensure you have a basic understanding of Docker to proceed.

> ## Celestia-App Setup with Docker

Reviewer: use Celestia-app in this heading

This guide provides instructions on how to use the Celestia `txsim` Docker image.

> ### Overview of Celestia-App `txsim`

Reviewer: what is this? ive never heard of it? is this ai-generated?

The Celestia-App `txsim` binary is a tool that can be
used to simulate transactions on the Celestia network.
It can be used to test the performance of the Celestia network.

> The txsim Docker image runs the `txsim` binary with multiple configurable options.

Reviewer: ????????

### Prerequisites to run Celestia-App on Docker

- Docker and a basic understanding of docker
- A prefunded account set up with the keyring stored in a file,
to be accessed by an instance of the docker image.
For more information on setting up a prefunded account,
refer to [key management](https://docs.celestia.org/nodes/celestia-app-commands).

### Celestia-App Quick-Start

1. In your local machine, navigate to the home directory

   ```bash
   cd $HOME
   ```

2. Create a file in which the keyring would be stored.
The file would be mounted as a volume into the docker container.

   ```bash
   touch $HOME/.celestia-app
   ```

3. Using a suitable text editor of your choice, open the

> .celestia-app file and paste the keyring of the prefunded account.

Reviewer: ????

4. We recommend that you set the necessary file permission for the
.celestia-app file. A simple read access is all that is required for the
docker container to access the content of the file.

5. You can run the txsim Docker image using the docker run command below.

   ```bash
   docker run -it \
   -v $HOME/.celestia-app:/home/celestia ghcr.io/celestiaorg/txsim \
   -k 0 \
   -g consensus-validator-robusta-rc6.celestia-robusta.com: \

>    -t 10s -b 10 -d 100 -e 10

Reviewer: ```suggestion
-B -S
```

   ```

6. In this command, the -v option is used to mount the
$HOME/.celestia-app directory from the host to the /home/celestia
directory in the Docker container.
This allows the txsim binary to access the keyring for the prefunded account.

Congratulations! You have successfully set up Celestia-App in Docker 😎

### Flag Breakdown

The table below provides a brief explanation of the
flags used in the docker run command in step 5 of the quick start instructions.

| FLAG | DESCRIPTION | DEFAULT | OPTION |
| ---- | ---- | ---- | :----: |

> |`-k`|Whether a new key should be created|0|1 for yes, 0 for no|

Reviewer: going to close this PR. this is all nonsense

   :::

> Congratulations! You now have a Celestia-Node running 😎

Reviewer: but i appreciate the emoji 🫡
